src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -263,9 +290,202 @@ public static void TestingNumbersInvalidConversionToInt64(string jsonString, dou
+                        json.GetUInt32();
+                        Assert.True(false, "Expected GetUInt32 to throw FormatException.");
+                    }
+                    catch (FormatException)
+                    {
+                        /* Expected exception */
+                    }
+                    doubleValue = json.GetDouble();
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("0.000", 0)]
+        [InlineData("1e1", 10)]
+        [InlineData("1.1e2", 110)]
+        [InlineData("12345.1", 12345.1)]
+        [InlineData("123456789012345678901", 123456789012345678901d)]
+        [InlineData("-1", -1)] // ulong.MinValue - 1
+        public static void TestingNumbersInvalidConversionToUInt64(string jsonString, double expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
+                    Assert.False(json.TryGetUInt64(out ulong value));
+                    Assert.True(json.TryGetDouble(out double doubleValue));
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+
+                    #region Get calls
+                    try
+                    {
+                        json.GetUInt64();
+                        Assert.True(false, "Expected GetInt64 to throw FormatException.");
+                    }
+                    catch (FormatException)
+                    {
+                        /* Expected exception */
+                    }
+                    doubleValue = json.GetDouble();
+                    Assert.Equal(expected, doubleValue);
+                    #endregion
+                }
+            }
+
+            Assert.Equal(dataUtf8.Length, json.BytesConsumed);
+            Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
+        }
+
+        [Theory]
+        [InlineData("-4.402823E+38", float.NegativeInfinity)] // float.MinValue - 1
+        [InlineData("4.402823E+38", float.PositiveInfinity)]  // float.MaxValue + 1
+        public static void TestingNumbersInvalidConversionToSingle(string jsonString, float expected)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes(jsonString);
+
+            var json = new Utf8JsonReader(dataUtf8, true, default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.Number)
+                {
+                    #region TryGet calls
+                    Assert.True(json.TryGetSingle(out float actual));
```

Comment:
https://mc.dot.net/#/user/WinCPP/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/adc5bf4dfda7358763b2eaeeb61a8eb676b16095/workItem/System.Text.Json.Tests/analysis/xunit/System.Text.Json.Tests.Utf8JsonReaderTests~2FTestingNumbersInvalidConversionToSingle(jsonString:%20%5C%224.402823E+38%5C%22,%20expected:%20%E2%88%9E)

This test fails for netfx since the `Utf8Parser.TryParse` did not support such numbers (and would return false rather than infinity). I would skip this test (and `TestingNumbersInvalidConversionToDouble`) for netfx.

```text
[SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Utf8Parser does not support parsing really large float and double values to infinity.")]
```

cc @tannergooding 

```text
Message :
Assert.True() Failure
Expected: True
Actual:   False
Stack Trace :
   at System.Text.Json.Tests.Utf8JsonReaderTests.TestingNumbersInvalidConversionToSingle(String jsonString, Single expected) in D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Text.Json\tests\Utf8JsonReaderTests.TryGet.cs:line 418
```